### PR TITLE
Ebpf swap

### LINKED ISSUE #11141: Extend SWAP monitoring
As it was initially explained on the issue https://github.com/netdata/product/issues/1940, we are bringing additional information related to swap memory using eBPF.

Our goal with this new feature is to show when and how processes are accessing the memory swap.

### PR DESCRIPTION
**Pull request by thiagoftsm**
##### Summary
Fixes #11141 

##### Component Name
ebpf.plugin
##### Test Plan

1 - Enable the swap monitoring adding the following line inside configuration file `/etc/netdata/ebpf.d.conf`:

```conf
[ebpf programs]
    swap = yes
``` 
2 - Compile this PR
3 - Force your environment to move more data from memory to swap running the following command:

```bash
echo 100 > /proc/sys/vm/swappiness
```

For more details about this action, you can read [this](https://www.howtoforge.com/tutorial/linux-swappiness/).

##### Additional Information
This PR was tested on the following environments:

| Distribution | Kernel |
|--------------|--------|
| Slackware current |  5.12.0 |
| Slackware current |  5.10.36 |
| Slackware current |  5.10.34 |
| Manjaro 21.0.2 |  5.10.32 |
| Ubuntu 18.04 |  5.4.0-72 |
| Ubuntu 18.04 |  4.15.0-142 |
| CentOS 7.9.2009 |  3.10.0-1160 |

### CODE REVIEW

File: web/gui/dashboard_info.js [hunk 1]
```diff
@@ -3486,6 +3490,16 @@ netdataDashboard.context = {
         info: 'Difference between the number of process created and the number of threads created per period(<code>process</code> dimension), it also shows the number of possible zombie process running on system.'
     },
 
+    'ebpf.swap_read_call': {
+        title : 'SWAP read page',
```
Comment (vlvkobal) [inline web/gui/dashboard_info.js:3494]:
```suggestion
    'apps.swap_read_call': {
```

File: web/gui/dashboard_info.js [hunk 2]
```diff
@@ -3486,6 +3490,16 @@ netdataDashboard.context = {
...
+        info: 'Monitor when a software read a page from memory swap.'
+    },
+
+    'ebpf.swap_write_call': {
+        title : 'SWAP write page',
```
Comment (vlvkobal) [inline web/gui/dashboard_info.js:3499]:
```suggestion
    'apps.swap_write_call': {
```

File: web/gui/dashboard_info.js [hunk 3]
```diff
@@ -863,6 +863,10 @@ netdataDashboard.context = {
         info: 'Total Swap I/O. (netdata measures both <code>in</code> and <code>out</code>. If either of the metrics <code>in</code> or <code>out</code> is not shown in the chart, the reason is that the metric is zero. - you can change the page settings to always render all the available dimensions on all charts).'
     },
 
+    'system.swapcalls': {
+        info: 'Total Swap Calls. (netdata measures calls for <code>write</code> and <code>read</code>. If either of the metrics <code>write</code> or <code>read</code> is not shown in the chart, the reason is that the metric is zero. - you can change the page settings to always render all the available dimensions on all charts). This chart is made for <code>ebpf.plugin</code>'
```
Comment (vlvkobal) [inline web/gui/dashboard_info.js:867]: > If either of the metrics <code>write</code> or <code>read</code> is not shown in the chart, the reason is that the metric is zero. - you can change the page settings to always render all the available dimensions on all charts). This chart is made for <code>ebpf.plugin</code>

I don't think we need that part at all.

**Review by vlvkobal — COMMENTED** [review]
Please consider removing changes from `dashboard_info.js` and using titles instead. It looks like you don't have any useful information to show in info strings. It would be nice if you discuss the titles with @joelhans.

File: collectors/ebpf.plugin/ebpf_swap.c [hunk 4]
```diff
@@ -0,0 +1,437 @@
...
+
+/**
+ * Sum PIDs
+ *
+ * Sum values for all targets.
+ *
+ * @param swap
+ * @param root
+ */
+static void ebpf_swap_sump_pids(netdata_publish_swap_t *swap, struct pid_on_target *root)
```
Comment (vlvkobal) [inline collectors/ebpf.plugin/ebpf_swap.c:230]: `ebpf_swap_sump_pids` -> `ebpf_swap_sum_pids`

**Comment by thiagoftsm** [comment]
Thank you @vlvkobal everything you required was done.

File: collectors/ebpf.plugin/ebpf_swap.c [hunk 5]
```diff
@@ -0,0 +1,437 @@
...
+ *
+ * @param em a pointer to the structure with the default values.
+ */
+void ebpf_swap_create_apps_charts(struct ebpf_module *em, void *ptr)
+{
+    UNUSED(em);
+
+    struct target *root = ptr;
+    ebpf_create_charts_on_apps(NETDATA_MEM_SWAP_READ_CHART,
+                               "Calls for function <code>swap_readpage.</code>",
```
[the remark this answers is not in the document]
Comment (thiagoftsm) [reply collectors/ebpf.plugin/ebpf_swap.c]: Thanks @vlvkobal, I pushed a commit with everything we talked about and fixes for the titles.

**Review by underhood — APPROVED** [review]

**Review by ilyam8 — APPROVED** [review]

**Review by vlvkobal — APPROVED** [review]

**Pull request merged**